src/main/java/world/bentobox/bentobox/hooks/LangUtilsHook.java

```diff
@@ -486,37 +485,41 @@ public static String getPredefinedTropicalFishName(TropicalFishBucketMeta meta,
-                    case 65536    : return "Clownfish";
-                    case 50726144 : return "Cotton Candy Betta";
-                    case 67764993 : return "Dottyback";
-                    case 234882305: return "Emperor Red Snapper";
-                    case 67110144 : return "Goatfish";
-                    case 117441025: return "Moorish Idol";
-                    case 16778497 : return "Ornate Butterflyfish";
-                    case 101253888: return "Parrotfish";
-                    case 50660352 : return "Queen Angelfish";
-                    case 918529   : return "Red Cichlid";
-                    case 235340288: return "Red Lipped Blenny";
-                    case 918273   : return "Red Snapper";
-                    case 67108865 : return "Threadfin";
-                    case 917504   : return "Tomato Clownfish";
-                    case 459008   : return "Triggerfish";
-                    case 67699456 : return "Yellowtail Parrotfish";
-                    case 67371009 : return "Yellow Tang";
-                }
-            } catch (NoSuchFieldException | IllegalAccessException ignored) {
-                // nothing
+            TropicalFish.Pattern pattern = meta.getPattern();
+
+            // https://minecraft.gamepedia.com/Tropical_Fish#Entity_data
+
+            int type = pattern.ordinal() > 5 ? 1 : 0;
+            int patt = pattern.ordinal() % 6;
+            int bcol = meta.getBodyColor().ordinal();
+            int pcol = meta.getPatternColor().ordinal();
+
+            int variant = (pcol & 255) << 24 | (bcol & 255) << 16 | (patt & 255) << 8 | type;
+
+            switch (variant) {
+                case 117506305: return "Anemone";
+                case 117899265: return "Black Tang";
+                case 185008129: return "Blue Tang";
+                case 117441793: return "Butterflyfish";
+                case 118161664: return "Cichlid";
+                case 65536    : return "Clownfish";
+                case 50726144 : return "Cotton Candy Betta";
+                case 67764993 : return "Dottyback";
+                case 234882305: return "Emperor Red Snapper";
+                case 67110144 : return "Goatfish";
+                case 117441025: return "Moorish Idol";
+                case 16778497 : return "Ornate Butterflyfish";
+                case 101253888: return "Parrotfish";
+                case 50660352 : return "Queen Angelfish";
+                case 918529   : return "Red Cichlid";
+                case 235340288: return "Red Lipped Blenny";
+                case 918273   : return "Red Snapper";
+                case 67108865 : return "Threadfin";
+                case 917504   : return "Tomato Clownfish";
+                case 459008   : return "Triggerfish";
+                case 67699456 : return "Yellowtail Parrotfish";
+                case 67371009 : return "Yellow Tang";
+                default       : return null;
             }
         }
         return null;
```

Comment:
If you are planning to return null, then please at @Nulllable annotation to the method.
Otherwise, someone could run into null pointer which they did not want.

Reply:
Yes, `@Nullable` is on line 481.

Reply:
Yes, the `@Nullable` annotation is on line 234.